jsonrpc/server.go

```diff
@@ -0,0 +1,279 @@
+// Package jsonrpc implements a JSONRPC2.0 compliant server as described in https://www.jsonrpc.org/specification
+package jsonrpc
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"reflect"
+	"strings"
+)
+
+const (
+	InvalidJson    = -32700 // Invalid JSON was received by the server.
+	InvalidRequest = -32600 // The JSON sent is not a valid Request object.
+	MethodNotFound = -32601 // The method does not exist / is not available.
+	InvalidParams  = -32602 // Invalid method parameter(s).
+	InternalError  = -32603 // Internal JSON-RPC error.
+)
+
+type request struct {
+	Version string `json:"jsonrpc"`
+	Method  string `json:"method"`
+	Params  any    `json:"params,omitempty"`
+	Id      any    `json:"id,omitempty"`
+}
+
+type batch []json.RawMessage
+
+type response struct {
+	Version string `json:"jsonrpc"`
+	Result  any    `json:"result,omitempty"`
+	Error   *Error `json:"error,omitempty"`
+	Id      any    `json:"id"`
+}
+
+type Error struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+	Data    any    `json:"data,omitempty"`
```

Comment:
This is a generic JSONRPC package, so starknet related comments shouldnt go here.

`Data` is produced by handlers and server just passes them along in the response. There is not much it can do with it anyways.